multitenant: replace EXPERIMENTAL_RELOCATE StoreID to NodeID lookup

ecwall opened the pull request:
Fixes #91628

Replace EXPERIMENTAL_RELOCATE's StoreID -> NodeID lookup's gossip impl with an in-memory cache.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/92709)
<!-- Reviewable:end -->

knz reviewed (commented):
I like it. But let's see what Arul thinks. 

> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-:-NI80cFJ3M4mICKRPB_T:bdfi13y)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @rafiss)



<!-- Sent from Reviewable.io -->

nvb reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @ecwall, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 435 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NI9DwR46y7yDkMAYjhP:-NI9DwR46y7yDkMAYjhQ:bs71j1e) ([raw file](https://github.com/cockroachdb/cockroach/blob/9f8aab453be4194077b577528118824ad43fe7c5/pkg/ccl/kvccl/kvtenantccl/connector.go#L435)):*
> ```Go
> }
> 
> // StoreIDToNodeID implements the serverpb.TenantStatusServer interface.
> ```

Did we consider mirroring the strategy that we use to proxy gossip subscriptions for `NodeDescriptor`s through to the storage node? See `Connector.updateNodeAddress` and `Connector.GetNodeDescriptor`. We could do the same for `StoreDescriptor`s by subscribing (`gossipSubsHandlers`) to `gossip.MakePrefixPattern(gossip.KeyStoreDescPrefix)`.


<!-- Sent from Reviewable.io -->

ecwall committed:
multitenant: replace EXPERIMENTAL_RELOCATE StoreID to NodeID lookup

Fixes #91628

Replace EXPERIMENTAL_RELOCATE's StoreID -> NodeID lookup's gossip
impl with an in-memory cache.

Release note: None

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 435 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NI9DwR46y7yDkMAYjhP:-NIEZrF08NvDwnQ5dK7x:biatmv0) ([raw file](https://github.com/cockroachdb/cockroach/blob/9f8aab453be4194077b577528118824ad43fe7c5/pkg/ccl/kvccl/kvtenantccl/connector.go#L435)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Did we consider mirroring the strategy that we use to proxy gossip subscriptions for `NodeDescriptor`s through to the storage node? See `Connector.updateNodeAddress` and `Connector.GetNodeDescriptor`. We could do the same for `StoreDescriptor`s by subscribing (`gossipSubsHandlers`) to `gossip.MakePrefixPattern(gossip.KeyStoreDescPrefix)`.
</blockquote></details>

I'm not familiar with this part of the code so I didn't think of that originally, but I updated it based on your suggestion which has simplified it a lot.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (changes requested):
> Reviewed 1 of 8 files at r1, 2 of 10 files at r2, 5 of 5 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIElGcmCZXU6eXVV8NY:-NIElGcmCZXU6eXVV8NZ:bhv3rup) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L150)):*
> ```Go
> 
> 	c.mu.nodeDescs = make(map[roachpb.NodeID]*roachpb.NodeDescriptor)
> 	c.mu.nodeDescsByStoreID = make(map[roachpb.StoreID]*roachpb.NodeDescriptor)
> ```

Did you consider making this a bit more generalizable and storing the entire `StoreDescriptor` here instead?
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 324 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEmczX3u6ywqYzrRj8:-NIEmczX3u6ywqYzrRj9:bouot0c) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L324)):*
> ```Go
> 	}
> 
> 	// TODO(nvanbenschoten): this doesn't handle StoreDescriptor removal from the
> ```

I haven't looked deeply into what that comment is saying, but to confirm, have we validated the same issue applies to store descriptors? 
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 317 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEng8c5g-QqqjcbT4s:-NIEng8c5g-QqqjcbT4t:b-lb3n20) ([raw file](https://github.com/cockroachdb/cockroach/blob/c04e23c43b0ed8a1b90370b750baebc368cf8445/pkg/ccl/kvccl/kvtenantccl/connector.go#L317)):*
> ```Go
> // updateStoreAddress handles updates to "store" gossip keys, performing the
> // corresponding update to the Connector's cached NodeDescriptor set.
> func (c *Connector) updateStoreAddress(ctx context.Context, key string, content roachpb.Value) {
> ```

Can we add a test to make sure this map is populated and we receive updates to `StoreDescriptors`? 
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 22 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEjR-T4bMrA-_cjQg1:-NIEjR-T4bMrA-_cjQg2:bn7fw2f) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/kv/kvclient/kvcoord/node_store.go#L22)):*
> ```Go
> 	// It returns an error if the node is not known by the store.
> 	GetNodeDescriptor(roachpb.NodeID) (*roachpb.NodeDescriptor, error)
> 	// GetNodeDescriptorByStoreID looks up the descriptor of the node by store ID.
> ```

Similar to my comment elsewhere, should we instead make this more generalizable and have it return a `StoreDescriptor` instead? 
___
*[`pkg/sql/relocate_range.go` line 58 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEnMKg1sGWunTknHHT:-NIEnMKh-CP4SZEtjIx1:b3tb2we) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/sql/relocate_range.go#L58)):*
> ```Go
> }
> 
> // todo(ewall): replace with NodeDescStore.GetNodeDescriptorByStoreID.
> ```

nit: "TODO(ewall)"


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIElGcmCZXU6eXVV8NY:-NIEvB-89iuOAGe71wnG:b-53kduv) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L150)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Did you consider making this a bit more generalizable and storing the entire `StoreDescriptor` here instead?
</blockquote></details>

I think that is more complicated because the `NodeDescriptor` inside the `StoreDescriptor` needs to be kept in sync when a `NodeDescriptor` updates. This is also true for the impl I currently have so I am changing this to a `map[roachpb.StoreID]roachpb.NodeID` and doing a double lookup in order to use `nodeDescs` as a single source of truth.

If at some point in the future we need to cache both `NodeDescriptor`s and `StoreDescriptor`s, we might need to think of how to keep the caches in sync.
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 324 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEmczX3u6ywqYzrRj8:-NIEw1d6Baw0ONxQ8C6t:b-1th0dr) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L324)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

I haven't looked deeply into what that comment is saying, but to confirm, have we validated the same issue applies to store descriptors? 
</blockquote></details>

Keys are only added to the map/cache, but never removed.
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 317 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEng8c5g-QqqjcbT4s:-NIEwNgCB-Re6zD59ZSk:blqb4q9) ([raw file](https://github.com/cockroachdb/cockroach/blob/c04e23c43b0ed8a1b90370b750baebc368cf8445/pkg/ccl/kvccl/kvtenantccl/connector.go#L317)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Can we add a test to make sure this map is populated and we receive updates to `StoreDescriptors`? 
</blockquote></details>

I'm not sure how to test this independently (`TestMultiTenantAdminFunction` tests this along with every other layer). Is there an existing more specific test for `updateNodeAddress`?
___
*[`pkg/sql/relocate_range.go` line 58 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEnMKg1sGWunTknHHT:-NIEv7y63VZKBb_pZTTD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/sql/relocate_range.go#L58)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: "TODO(ewall)"
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIElGcmCZXU6eXVV8NY:-NIFUnExB-uTIq0OZIOA:b510sf8) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L150)):*
> we might need to think of how to keep the caches in sync.

Is this possible? We Gossip store descriptors around here -- https://github.com/cockroachdb/cockroach/blob/830ab32a264394943bbded3f8bf16994f3e400a4/pkg/kv/kvserver/store.go#L2598-L2605

Tracing through that `s.Descriptor` call, it seems like this is set from the `NodeDescriptor` on the `Node` struct. My reading of this field is that it's immutable, so they should never be out of sync, by design. 

There's benefit to storing the entire `StoreDescriptor` here. One that comes to mind is this special casing we do around zone config validation for secondary tenants. We'd no longer need to make a distinction here between the system tenant and secondary tenants if we had access to the entire `StoreDescriptor`. 

https://github.com/cockroachdb/cockroach/blob/830ab32a264394943bbded3f8bf16994f3e400a4/pkg/sql/set_zone_config.go#L1003-L1015
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 317 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEng8c5g-QqqjcbT4s:-NIFgtmu9PWJZHd7lW3U:bookmf) ([raw file](https://github.com/cockroachdb/cockroach/blob/c04e23c43b0ed8a1b90370b750baebc368cf8445/pkg/ccl/kvccl/kvtenantccl/connector.go#L317)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

I'm not sure how to test this independently (`TestMultiTenantAdminFunction` tests this along with every other layer). Is there an existing more specific test for `updateNodeAddress`?
</blockquote></details>

I was looking around and I came across `NewStoreGossiper`. Maybe we can make use of that to trigger gossip updates and make sure we're able to see them?


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 317 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEng8c5g-QqqjcbT4s:-NIJ-4Gi3-oDfO6EHa_k:b-hqvbce) ([raw file](https://github.com/cockroachdb/cockroach/blob/c04e23c43b0ed8a1b90370b750baebc368cf8445/pkg/ccl/kvccl/kvtenantccl/connector.go#L317)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

I was looking around and I came across `NewStoreGossiper`. Maybe we can make use of that to trigger gossip updates and make sure we're able to see them?
</blockquote></details>

Added some assertions to `TestConnectorGossipSubscription`.


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIElGcmCZXU6eXVV8NY:-NIJobFe4ITruQWQRQuA:b1we9td) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/ccl/kvccl/kvtenantccl/connector.go#L150)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

> we might need to think of how to keep the caches in sync.

Is this possible? We Gossip store descriptors around here -- https://github.com/cockroachdb/cockroach/blob/830ab32a264394943bbded3f8bf16994f3e400a4/pkg/kv/kvserver/store.go#L2598-L2605

Tracing through that `s.Descriptor` call, it seems like this is set from the `NodeDescriptor` on the `Node` struct. My reading of this field is that it's immutable, so they should never be out of sync, by design. 

There's benefit to storing the entire `StoreDescriptor` here. One that comes to mind is this special casing we do around zone config validation for secondary tenants. We'd no longer need to make a distinction here between the system tenant and secondary tenants if we had access to the entire `StoreDescriptor`. 

https://github.com/cockroachdb/cockroach/blob/830ab32a264394943bbded3f8bf16994f3e400a4/pkg/sql/set_zone_config.go#L1003-L1015
</blockquote></details>

Updated based on our meeting.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (commented):
Thanks for making the change! I really like that this will help us add the same zone config validation for secondary tenants that only exists for the system tenant today. 

This should be good to merge once we've reverted the rename of `NodeDescStore` to `DescCache`. 

> Reviewed 18 of 30 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-:-NIJvHlj83l82k_QPGxe:bhs5xdx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 317 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEng8c5g-QqqjcbT4s:-NIJryyq1RtwIdESeyIX:b-83sjoh) ([raw file](https://github.com/cockroachdb/cockroach/blob/c04e23c43b0ed8a1b90370b750baebc368cf8445/pkg/ccl/kvccl/kvtenantccl/connector.go#L317)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

Added some assertions to `TestConnectorGossipSubscription`.
</blockquote></details>

Nice!
___
*[`pkg/gossip/gossip.go` line 897 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtOFk4lIga10QzKfP:-NIJtOFk4lIga10QzKfQ:bbq0u1e) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/gossip/gossip.go#L897)):*
> ```Go
> 
> // updateStoreAddress is a gossip callback which is used to update storeMap.
> func (g *Gossip) updateStoreAddress(key string, content roachpb.Value) {
> ```

I don't think we're updating any addresses here. Should we switch back to `updateStoreMap` instead? 
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 18 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtanY2YO0wkn0W_Rt:-NIJtanY2YO0wkn0W_Ru:bqmphkg) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvcoord/node_store.go#L18)):*
> ```Go
> //
> // Implementations of the interface are expected to be threadsafe.
> type DescCache interface {
> ```

Can we revert this change? `DescCache` is too generalized given how overloaded the term is. I think it's fine if `StoreDescriptors` get lumped under `NodeDescsStore`. 
___
*[`pkg/kv/kvclient/kvtenant/connector.go` line 50 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJuGij4j0LRZUT7i7-:-NIJuGij4j0LRZUT7i70:b45dhyg) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvtenant/connector.go#L50)):*
> ```Go
> 
> 	// DescCache provides information on each of the KV nodes in the cluster
> 	// in the form of NodeDescriptors. This obviates the need for SQL-only
> ```

"in the form of NodeDescriptors and StoreDescriptors" 
___
*[`pkg/sql/exec_util.go` line 1345 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJuXXF6s2THfkxttDF:-NIJuXXF6s2THfkxttDG:b-4owie4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/sql/exec_util.go#L1345)):*
> ```Go
> 	EventsExporter obs.EventsExporter
> 
> 	// DescCache stores node descriptors in an in-memory cache.
> ```

"stores {Store,Node]Descriptors .. "


<!-- Sent from Reviewable.io -->

knz reviewed (commented):
> Reviewed 1 of 5 files at r3, 28 of 30 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @ecwall, @nvanbenschoten, and @rafiss)

___
*[`pkg/sql/exec_util.go` line 1345 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJuXXF6s2THfkxttDF:-NIXb4LpBwzJ6Gt_IPK-:b7a91fi) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/sql/exec_util.go#L1345)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

"stores {Store,Node]Descriptors .. "
</blockquote></details>

I think you meant a `}` here not `]`
___
*[`pkg/sql/relocate_range.go` line 58 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIXbjXv7_W68kTUzCib:-NIXbjXwCvxJsHThu5oW:bvxg8x7) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2826a3a609da497d0ad899ae74fed08189af176/pkg/sql/relocate_range.go#L58)):*
> ```Go
> }
> 
> // TODO(ewall): replace with DescCache.GetStoreDescriptor.
> ```

I think it is time to do so?


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/gossip/gossip.go` line 897 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtOFk4lIga10QzKfP:-NIOARlCEmTO5zhvSkcu:bjfskzk) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/gossip/gossip.go#L897)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

I don't think we're updating any addresses here. Should we switch back to `updateStoreMap` instead? 
</blockquote></details>

There is no longer a `storeMap` field so I don't think there should be a `updateStoreMap` function.

The corresponding `NodeDescriptor` function is called `updateNodeAddress`, so I made it consistent with that.
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 22 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIEjR-T4bMrA-_cjQg1:-NIOCHsl1s9d_X7MjUQK:b-vvwx6j) ([raw file](https://github.com/cockroachdb/cockroach/blob/63254f231303242d6206363b2a3114eda4396e0c/pkg/kv/kvclient/kvcoord/node_store.go#L22)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Similar to my comment elsewhere, should we instead make this more generalizable and have it return a `StoreDescriptor` instead? 
</blockquote></details>

Replied in other thread.
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 18 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtanY2YO0wkn0W_Rt:-NIOBJll9nbQA36J4AWh:bqs05pr) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvcoord/node_store.go#L18)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Can we revert this change? `DescCache` is too generalized given how overloaded the term is. I think it's fine if `StoreDescriptors` get lumped under `NodeDescsStore`. 
</blockquote></details>

I think `NodeDescsStore` is confusing for 2 reasons
1) It is now storing StoreDescriptors as much as it is storing NodeDescriptors. Why not call it StoreDescStore?
2) "Store" is now overloaded to mean both StoreDescriptor and an in-memory data store.
___
*[`pkg/kv/kvclient/kvtenant/connector.go` line 50 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJuGij4j0LRZUT7i7-:-NIOCTma1tONfkvnwxLo:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvtenant/connector.go#L50)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

"in the form of NodeDescriptors and StoreDescriptors" 
</blockquote></details>

Fixed.
___
*[`pkg/sql/exec_util.go` line 1345 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJuXXF6s2THfkxttDF:-NIOCfh91vMJgPpJU7Kl:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/sql/exec_util.go#L1345)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

"stores {Store,Node]Descriptors .. "
</blockquote></details>

Fixed.
___
*[`pkg/sql/relocate_range.go` line 58 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIXbjXv7_W68kTUzCib:-NIXldaVDS-F-E1iv99e:b-5al1qb) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2826a3a609da497d0ad899ae74fed08189af176/pkg/sql/relocate_range.go#L58)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

I think it is time to do so?
</blockquote></details>

I have a separate PR for this.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/gossip/gossip.go` line 897 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtOFk4lIga10QzKfP:-NIXpIt-Dr0UQVnrL3Bt:boo4xae) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/gossip/gossip.go#L897)):*
> The corresponding NodeDescriptor function is called updateNodeAddress, so I made it consistent with that.

Well, that's because that function actually updates node addresses. There's no addresses to speak of when we're dealing with `StoreDescriptors`, unlike `NodeDescriptors`, so I'm not sure what being consistent with `updateNodeAddresses` gives us. 

I buy that you're changing how the map is stored, but it's still a map at the end of the day. 
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 18 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtanY2YO0wkn0W_Rt:-NIXqmYr0mZJfzf6JSo0:bmsmes0) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvcoord/node_store.go#L18)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

I think `NodeDescsStore` is confusing for 2 reasons
1) It is now storing StoreDescriptors as much as it is storing NodeDescriptors. Why not call it StoreDescStore?
2) "Store" is now overloaded to mean both StoreDescriptor and an in-memory data store.
</blockquote></details>

Sure, but it isn't obvious to me what a `kvcoord.DescCache` stores given how overloaded/generalizable those two words are.  I'd wager other readers would have the same problem. 

Can we call this something more specific, if you don't want to revert the change?


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/gossip/gossip.go` line 897 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtOFk4lIga10QzKfP:-NIY6e7G6_jwI4CoFh2V:b2qaecx) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/gossip/gossip.go#L897)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

> The corresponding NodeDescriptor function is called updateNodeAddress, so I made it consistent with that.

Well, that's because that function actually updates node addresses. There's no addresses to speak of when we're dealing with `StoreDescriptors`, unlike `NodeDescriptors`, so I'm not sure what being consistent with `updateNodeAddresses` gives us. 

I buy that you're changing how the map is stored, but it's still a map at the end of the day. 
</blockquote></details>

It looks like it updates the NodeDesc in general, not just the address. It just has some special logic around writing the address specifically.
___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 18 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtanY2YO0wkn0W_Rt:-NIY6wyQDBJhlxbq3vwE:bmnssbd) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvcoord/node_store.go#L18)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Sure, but it isn't obvious to me what a `kvcoord.DescCache` stores given how overloaded/generalizable those two words are.  I'd wager other readers would have the same problem. 

Can we call this something more specific, if you don't want to revert the change?
</blockquote></details>

I'm not sure what else to call it since I thought it would cache any Descriptors from KV so I'll just revert it.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 30 of 30 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-:-NIYHqpiA4isExeIc3sb:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ecwall, @knz, @nvanbenschoten, and @rafiss)

___
*[`pkg/kv/kvclient/kvcoord/node_store.go` line 18 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/92709#-NIJtanY2YO0wkn0W_Rt:-NIYGyvD1FVZGU4Wmoyx:bg6g6bu) ([raw file](https://github.com/cockroachdb/cockroach/blob/5237cc85f999a55742c4a5cdea7f67af009181f5/pkg/kv/kvclient/kvcoord/node_store.go#L18)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

I'm not sure what else to call it since I thought it would cache any Descriptors from KV so I'll just revert it.
</blockquote></details>

Thanks for reverting. As an example, we would never expect this to cache things like `RangeDescriptors`, which is why the overloaded term brought about more questions than it answered. 


<!-- Sent from Reviewable.io -->
